### PR TITLE
Allow the owner to reset DANK for a phone number and the allotment

### PR DESCRIPTION
**Pull request by danielglauser**

### CODE REVIEW

File: contracts/DankFaucet.sol [hunk 1]
```diff
@@ -104,4 +108,20 @@ contract DankFaucet is usingOraclize {
    * Simple function to allow for adding ETH to the contract.
    */
   function sendEth() public payable { }
+
+  /**
+   * Allow the owner to reset the DANK we've allocated to a phone number.
+   */
+  function resetAllocatedDankForPhoneNumber(bytes32 hashedPhoneNumber, uint dank) auth {
+    uint dankInWei = (dank * 1000000000000000000);
```
Comment (jpmonettas) [inline contracts/DankFaucet.sol]: I think is better if the contract api accepts all values in wei, and the contracts only works in wei, so no conversion or weird multiplications needed in the contract.

File: contracts/DankFaucet.sol [hunk 2]
```diff
@@ -104,4 +108,20 @@ contract DankFaucet is usingOraclize {
    * Simple function to allow for adding ETH to the contract.
    */
   function sendEth() public payable { }
+
+  /**
+   * Allow the owner to reset the DANK we've allocated to a phone number.
+   */
+  function resetAllocatedDankForPhoneNumber(bytes32 hashedPhoneNumber, uint dank) auth {
```
Comment (jpmonettas) [inline contracts/DankFaucet.sol]: What is the purpose of the dank parameter here if we can only assign allotment amounts to users? Isn't it better to remove that parameter and just remove the phone entry in the map?

Comment (danielglauser) [reply contracts/DankFaucet.sol]: I went back and forth on this one. It was pretty easy to add in the amount that way we had more control with just a little bit of flexibility. That flexibility would only be useful in the case where we change the initial allotment and changed the logic to allow folks to "top off" once that allotment had changed. It doesn't make much sense right now so I can remove that argument and just remove the entry from the map.

File: contracts/DankFaucet.sol (hunk 1 above)
Comment (danielglauser) [reply contracts/DankFaucet.sol]: I stuck with the convention in the rest of the contract to take values in and multiply to get wei. Do you think I should remove this in both places in the contract?

Comment (jpmonettas) [reply contracts/DankFaucet.sol]: Yeah I think it is better to have everything in wei inside contracts.

Comment (madvas) [reply contracts/DankFaucet.sol]: yes, everything in wei inside contracts please

**Comment by madvas** [comment]
Now that you changed expected value to wei, doesn't it require changes to files outside smart-contracts as well?

**Comment by danielglauser** [comment]
@madvas  just ran the Truffle migration script and it’s transferring 10000000000000000 DANK to the faucet. Looks like we're good to go.

**Pull request merged**